db/order.py

```
@@ -0,0 +1,43 @@
+from django.db import transaction
+from django.db.models import QuerySet
+from django.contrib.auth import get_user_model
+from typing import List, Optional, Dict, Any
+import datetime
+
+from db.models import Ticket, Order, MovieSession
+
+
+@transaction.atomic
+def create_order(
+    tickets: List[Dict[str, Any]],
+    username: str,
+    date: Optional[str] = None,
+) -> Order:
+    """Create an order with associated tickets for a user."""
+    user = get_user_model().objects.get(username=username)
+    created_at = (
+        datetime.datetime.strptime(date, "%Y-%m-%d %H:%M")
+        if date
+        else datetime.datetime.now()
```

Issue: If the 'date' parameter does not match the expected format '%Y-%m-%d %H:%M', a ValueError will be raised. This error is not handled, which can cause runtime exceptions. Please add error handling or input validation to gracefully manage invalid date formats.